### DISCUSSION TITLE
wprig component 'initialize' vs WP 'init' hook

### DISCUSSION THREAD

**Discussion by petko** [wprig/wprig#824] [Q&A] [answered]
Is calling code in the wprig component `initialize` method equivalent to calling a code in the `init` hook in vanilla Wordpress? I need to call some code in the `init` hook and I was wondering if explicitly adding this hook might be redundant?

**Comment by dthenley** [comment, marked as answer]
Let me know if I'm off the mark on what you are wanting to do here.

If you are wanting to call the `init` hook in wp rig you will call it in the `initialize` method. So it's not really the same but it is the location you would want to call the hook.

So for example the "proper" way to do it in rig is 

```
public function initialize() {
    add_action( 'init', array( $this, 'your_custom_function' ) );
}

public function your_custom_function() {
return  value;
}
```
You can view more on how to create a WP Rig component here https://wprig.io/learn/functions-in-wp-rig-v2/.

**Reply by petko** [reply]
Yes, I was wondering if both are the same, because of the name resemblance. I know that I should call the `add_action` functions in `initialize` generally. Thanks!

**Reply by dthenley** [reply]
Ah ok gotcha! Yeah the initialize method is being called from inc/theme.php folder, it's not being directly called from wordpress. Glad I could help answer that for you.

**Comment by robruiz** [comment]
Just hoping to add a bit more context here in case others read this and have the same question. As with any other function that is part of a theme, everything stems from functions.php. When the theme files load, then the rest loads. Initialize is essentially the class constructor, so it gets fired when the class is "newed" up from the Theme.php file in /inc. The hook stack is kind of a whole different beast, if you are wondering about order of operations here. There's not much difference between adding your add_action in that constructor of the component and adding it right in the theme function.php file in the theme root. Just remember to new up your custom components at the bottom of that Theme.php file, which is basically where all of the theme classes get kicked off from.

**Reply by doughamlin** [reply]
What was the reason the use initialize vs __construct?
